### PR TITLE
build: add .d.ts

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * import isEs5IdentifierName from "@prettier/is-es5-identifier-name";
+ *
+ * isEs5IdentifierName("fisker");
+ * // -> true
+ *
+ * isEs5IdentifierName("16");
+ * // -> false
+ * ```
+ */
+declare function isEs5IdentifierName(id: string): boolean;
```
Comment (fisker) [inline src/index.d.ts]: Should this be exported?

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ *
+ * @example
+ * ```js
+ * import isEs5IdentifierName from "@prettier/is-es5-identifier-name";
+ *
+ * isEs5IdentifierName("fisker");
+ * // -> true
+ *
+ * isEs5IdentifierName("16");
+ * // -> false
```
Comment (fisker) [inline src/index.d.ts]: Let's use `isEs5IdentifierName("prettier");` as example (sync readme.md please).  I think one case is good enough.

**Comment by fisker** [comment]
Out of curiosity, are you going to use this package somewhere? If not, I don't think `.d.ts` file is necessary, since it only used in Prettier, and it's obvious that it accept a string returns a boolean.

**Comment by xiaoxiangmoe** [comment]
@fisker I want to remove this line https://github.com/prettier/prettier/blob/6e5dee0a0154b03a9a3318b41603c8abc9c565b7/tsconfig.json#L11

So I need add types for prettier's dependencies

**Comment by fisker** [comment]
Released https://github.com/prettier/is-es5-identifier-name/releases/tag/v0.2.0

**Pull request merged**